lib/_stream_readable.js

```diff
@@ -447,7 +447,8 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new Error('not implemented'));
+  const err = new Error('not implemented');
+  process.nextTick(() => this.emit('error', err));
 };
```

Comment:
Oh, I missed the fact your comment addressed all the function ode above and not just `_read`.

So - move it to a named function that takes the error as a parameter to avoid the closure?